### PR TITLE
Video listing page

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features added
- All videos listing page
- Filter videos by category

# demo
https://user-images.githubusercontent.com/54719132/172896637-dbd7be7b-580a-4821-b252-72aeca59a52c.mp4

### CODE REVIEW

**Review by Kavita539 — APPROVED** [review]
- code looks good and clean
- Folders are structured nicely
- UI looks good too
- Overall looks good, can merge it

File: src/Pages/VideoListingPage.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    const getVideos = async () => {
+        try {
+            const response = await axios.get("/api/videos");
+            setVideos(response.data.videos);
+        } 
+        catch (error) {
+            console.log(error);
+        }
+    }
```
Comment (shrushti2000) [inline src/Pages/VideoListingPage.jsx:30]: Just a suggestion, video functionality functions can be moved to other file seperately and can be named videoServices.js or something. Just a minor suggestion .

**Review by shrushti2000 — APPROVED** [review]
Overall looks good.  Left a minor suggestion. Can be merged. Good work

File: src/Pages/VideoListingPage.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <div className="video-listing-page">
+            <Navbar/>
+            <main id="homepage">
+
+                <div className="category-tab">
+                    <span className={`category-chip text-center ${selectedCategory === "All" ? "selected" : ""}`} value="All" onClick={(e) => setSelectedCategory(e.target.innerText) }>All</span>
+                    {categories.map(({categoryName, id}) => {
+                        return <span 
+                            key={id} 
+                            className={`category-chip text-center ${selectedCategory === categoryName ? "selected" : ""}`}
```
Comment (Shailseen) [inline src/Pages/VideoListingPage.jsx:63]: you can also simplify with **&&** for one condition only.

**Review by Shailseen — APPROVED** [review]
only one suggestion and also you can make service seprate for api calls.
In ui you can set category names instead of *All Videos** only suggestion.
Folder structure is good
You can go for merge

**Pull request merged**